resteasy-core/src/main/java/org/jboss/resteasy/plugins/providers/MultiValuedCollectionParamConverter.java

```diff
@@ -0,0 +1,65 @@
+package org.jboss.resteasy.plugins.providers;
+
+
+import java.lang.reflect.Constructor;
+
+import java.util.Collection;
+import javax.annotation.Priority;
+import javax.ws.rs.ext.ParamConverter;
+import javax.ws.rs.ext.Provider;
+
+import org.jboss.resteasy.core.StringParameterInjector;
+import org.jboss.resteasy.resteasy_jaxrs.i18n.Messages;
+
+/**
+ * @author Marek Kopecky mkopecky@redhat.com
+ * @author Ron Sigal rsigal@redhat.com
+ */
+@Provider
+@Priority(Integer.MAX_VALUE)
```

Comment:
https://github.com/resteasy/Resteasy/pull/1721/files#diff-505f9bfa494b22e39f4249e79a979f41R811
```
These built-in ParamConverters have the lowest priority ...
```
MIN_VALUE would be probably better, right? (if I don't missed something ...)

Anyway, I believe that `@Provider` and `@Priority` annotation shouldn't be in corverter classes (same for `MultiValuedArrayParamConverter` class), but in MultiValuedParamConverterProvider class, right? I see this approach for example in [PriorityEqualityTest](https://github.com/resteasy/Resteasy/blob/master/testsuite/unit-tests/src/test/java/org/jboss/resteasy/test/providers/PriorityEqualityTest.java#L79)

Reply:
> Anyway, I believe that @Provider and @Priority annotation shouldn't be in corverter classes (same for MultiValuedArrayParamConverter class), but in MultiValuedParamConverterProvider class, right?

I agree with you on this point.

> MIN_VALUE would be probably better, right? (if I don't missed something ...)

I worked on this point in another PR and if I 'm not wrong, I can say that to achieve following behavior:

> These built-in ParamConverters have the lowest priority, so any user supplied  ParamConverters will be tried first

the `@Priority(Integer.MAX_VALUE)` on built-in provider will not be taken into account.
 Instead this behavior will be achieved by the `ResteasyProviderFactoryImpl.ExtSortedKey.compareTo()` method that says that user supplied  `ParamConverters `are less than built-in `ParamConverters`.

`@Priority(Integer.MAX_VALUE)` on built-in provider will only be used to sort built-in `ParamConverters` between them. And the provider with the lowest priority is used first.

Reply:
@marekkopecky, @NicoNes, you're both right. I took out all of the @Priority annotations. Thank you! This is why it's good to be surrounded by smart people. ;-)